src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProjectTracker.cs

```diff
@@ -215,9 +215,12 @@ internal void AddProject(AbstractProject project)
 
         internal void StartPushingToWorkspaceAndNotifyOfOpenDocuments(IEnumerable<AbstractProject> projects)
         {
-            foreach (var hostState in _workspaceHosts)
+            using (Dispatcher.CurrentDispatcher.DisableProcessing())
```

Comment:
@jasonmalinowski and I talked in person, but this require `hostState`, where `NotifyWorkspaceHosts()` only gives the `host`.